### DISCUSSION TITLE
SWC file node -> compartment assignment behavior

### DISCUSSION THREAD

**Discussion by VENOM314** [jaxleyverse/jaxley#547] [General]
After communication with one of Jaxley's developers Michael, here is my brief understanding of the assignment behavior.

Jaxley calles a series of nodes with no branching a "branch". For example, [branching-3-4-5-6-branching], [3-4-5-6] is considered a branch. If we specify ncomp = 1, generally one branch corresponds to one compartment. Thus in general, compartment is not just [node1-node2], but _a series of nodes_ ! It is not direct mapping between SWC file and compartment

Furthermore, branches might be interrupted. Quoting Michael's words:

"This happens when the swc files traces a branch with interruptions from other branches (e.g. the person tracing the neuron traces branch 5, but then stops to trace branch 28, and then returns to trace branch 5. In that case, the “5” branch will be split into two subbranches in Jaxley). We might change this behavior in the future."

So it is quite hard to trace the assignment behavior by writing a "reverse-engineering algorithm" by oneself.

We can, however, use cell.xyzr[] function. This attribute saves all the swc coordinates by the branch that they are assigned to. For example, `cell.xyzr[5]` will show you all SWC coordinates that are merged into `cell.branch(5)`. One can also map .xyzr to SWC file by mapping nodes with exactly same coordinate (if one hopes to get the node-to-node topology/parent-children relationship).

**Comment by michaeldeistler** [comment]
Thanks for writing this up! One clarification to anybody reading this: In the text above, the word `node` refers to one line in the SWC file.

All the best
Michael

**Reply by VENOM314** [reply]
Right, one node is one line in SWC file, thank you for the clarification. When I was reverse engineering the assignment problem, I wrote a BFS algorithm that assigns nodes to compartments. What is Jaxley's algorithm and why might be a deeper reason for compartments being interrupted?

**Reply by michaeldeistler** [reply]
There is no deeper reason. As I also said already via email, we might even change the behavior of "interrupted compartments" in the future.
